### ISSUE TITLE
Suggestion: Reversed Portal

### ISSUE THREAD

**Issue by mcm-odoo** [odoo/owl#849]
Could be nice to have an utility component that works like a portal but in an opposite way
This component will catch events and redirect them to a component given in props.

I think this component could be nice with wowl's DialogManager/PopoverManager.
We could trigger an event in the dialog/popover and catch it in the view that "owns" the dialog/popover.

```xml
<t t-name="DialogManager">
  <div>
    <t t-foreach="components" t-as="component">
      <ReversedPortal redirectTo="ownerComponent">
        <t t-component="component" />
      </ReversedPortal>
    </t>
  </div>
</t>
```

At the moment, view has to give functions in props and the dialog/popover should call them.

**Comment by mcm-odoo** [comment]
Quick implementation I've made in Odoo

```js
const outPortalSymbol = Symbol("outPortal");
patch(Component.prototype, "OutPortal", {
  __trigger(component, evType, payload) {
    if (this.env[outPortalSymbol]) {
      this.env[outPortalSymbol](evType);
    }
    this._super(component, evType, payload);
  }
});

class OutPortal extends Component {
  setup() {
    useSubEnv({
      [outPortalSymbol]: this.addEventListener.bind(this),
    });
    this.handledEvents = new Set();
  }
  willUnmount() {
    this.handledEvents.clear();
  }
  addEventListener(evType) {
    if (this.el && !this.handledEvents.has(evType)) {
      this.handledEvents.add(evType);
      this.el.addEventListener(evType, (ev) => {
        this.props.owner.__trigger(ev.originalComponent, evType, ev.detail);
      });
    }
  }
}
OutPortal.template = xml`<t t-slot="default" />`;
```

**Comment by ged-odoo** [comment]
what's wrong with `At the moment, view has to give functions in props and the dialog/popover should call them.`?

**Comment by ged-odoo** [comment]
@mcm-odoo

**Comment by mcm-odoo** [comment]
It's not natural.

At first, I wanted to trigger an event (as we would do in any other component) then I remembered that the popover is not a child of the view.
So I used the props but it's not natural (for me) to have props that do the role of a trigger.
When we are in the popover, we would think there is something above it but there is nothing except the service's container.
An other thing is that popovers and dialogs are design both to be used directly in the template or with a call to their service so
we'll need to think twice in which context we are and remember which method we should use (a function prop or a trigger call).
We should not have to choose between these 2 methods, the trigger method is the natural way...

Anyway, after my last comment, I implemented it differently (for popover at least) by giving the handlers not to the popover with props but by giving the handlers to the service which then manages them.
